doc/userguide/rules/intro.rst

```diff
@@ -349,3 +349,189 @@ reassembled streams, TLS-, SSL-, SSH-, FTP- and dcerpc-buffers.
+    * - Decode Events Only
+      - Packet
+      - Per-packet basis
+      - Packets that are broken on an IP level
+      - decode-event
+    * - Packet
+      - Packet
+      - Per-packet basis
+      - Packet-level info (e.g.: header info)
+      - ttl
+    * - IP Only
+      - Flow
+      - Once per direction
+      - On IP addresses on the flow
+      - ---
+    * - IP Only (negated address)(*)
+      - Flow
+      - Once per direction
+      - On the flow, on IP address level (negated addressed)
+      - ---
+    * - Protocol Detection Only
+      - Flow
+      - Once per direction, when protocol detection is done
+      - On protocol detected for the flow
+      - app-layer-protocol
+    * - Packet-Stream
+      - Flow, if stateful (**)
+      - Flow, if stateful, per-packet if not
+      - Against the reassembled stream. If stream unavailable, match per-packet
+        (packet payload and stream payload)
+      - 'content' with 'starts with' or 'depth'
+    * - Stream
+      - Flow, if stateful (**)
+      - Per stream chunk, if stateful, per-packet if not
+      - Against the reassembled stream. If stream unavailable, match per-packet
+      - 'tcp-stream' in protocol field; simple 'content'; 'byte_extract'
+    * - Application Layer Protocol
+      - Flow
+      - Once per flow
+      - On 'protocol' field
+      - protocol field of a rule
+    * - Application Layer Protocol Transactions
+      - Flow
+      - Per reassembled stream segment
+      - On buffer keywords
+      - Application layer protocol-related, e.g. `http.host`, `rfb.secresult`,
+        dcerpc_stub_data
+
+.. note::
+    (*) IP Only signatures with negated addresses are _like_ IP-only signatures, but
+    currently handled differently due to limitations of the algorithm processing
+    IP Only rules.
+
+.. note:: Scope: `Flow, if stateful`
+
+    (**) Apply to the stream, if the reassembled stream is available. If
+    `stream-depth` is reached, or the stream is unreachable some other way, the
+    rule will be applied on a packet level.
+
+Signature Examples per Type
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Decode Events Only
+^^^^^^^^^^^^^^^^^^
+
+For more examples check https://github.com/OISF/suricata/blob/master/rules/decoder-events.rules.
+
+.. container:: example-rule
+
+    alert pkthdr any any -> any any (msg:"SURICATA IPv4 malformed option"; :example-rule-emphasis:`decode-event:ipv4.opt_malformed;` classtype:protocol-command-decode; sid:2200006; rev:2;)
+
+Packet
+^^^^^^
+
+.. container:: example-rule
+
+    alert udp any any -> any any (msg:"UDP with flow direction"; flow:to_server; sid:1001;)
+
+.. container:: example-rule
+
+    alert tcp any any -> any any (msg:"ttl"; :example-rule-emphasis:`ttl:123;` sid:701;)
+
+IP Only
+^^^^^^^
+
+.. container:: example-rule
+
+    alert tcp-stream any any -> any any (msg:"tcp-stream, no content"; sid:101;)
+
+
+.. container:: example-rule
+
+    alert tcp-pkt any any -> any any (msg:"tcp-pkt, no content"; sid:201;)
```

Comment:
Love these example rules! <3
Suggestion: Perhaps explicit IP addresses here

Reply:
Thanks, most of them are from the SV tests for rule types :D
I'll add a rule with explicit IP addresses, merci!